gpt_index/readers/string_iterable.py

```diff
@@ -0,0 +1,39 @@
+from abc import abstractmethod
+from typing import Any, Iterable, List
+
+from gpt_index.readers.base import BaseReader
+from gpt_index.readers.schema.base import Document
+
+
+class StringIterableReader(BaseReader):
+    """String Iterable Reader
+
+    Gets a list of documents, given an iterable (e.g. list) of strings
+
+    Args:
+        texts (Iterable[str]): Iterable of strings to be added to the Document list result.
+
+    Example:
+        .. code-block:: python
+            from gpt_index import StringIterableReader, GPTTreeIndex
+
+            documents = StringIterableReader(["I went to the store", "I bought an apple"]).load_data()
+            index = GPTTreeIndex(documents)
+            index.query("what did I buy?")
+
+            # response should be something like "You bought an apple."
+    """
+
+    def __init__(self, texts: Iterable[str]):
```

Comment:
I could have added this param to the load_data method, but it would have been harder to add an explicit type annotation for that

Reply:
I know passing parameters through load_kwargs is jank but I think that might be better. The init in the future can contain parameters like "exclude_char", "separator" etc but I think the data-specific stuff should be in the params on load_data.

Reply:
thanks! done https://github.com/jerryjliu/gpt_index/pull/173/commits/01793103f32891d6db2598cbd6562db1de555951